[#9508] Fix Hive SerDe incompatibility for Gravitino Flink connector created tables

Pranaykarvi opened the pull request:
### What changes were proposed in this pull request?

This PR fixes an interoperability issue between the Gravitino Flink connector and
the native Flink Hive client.

1. seperate properteis converter interface to catalog properteis converter  and schema&table properteis converter
2. create a hive schema&table properteis converter, to generate gravitino tables according to the table properties and hive conf to follow flink behavior
3. transform format, serde, input/output format
4. get serde by following rows:
```
  // 1. use the serde lib in the stored-as format
  // 2. use the serde lib specified in the properties
  // 3. use the serde lib from default file format
  // 4. use the default serde in hive conf
  // please refer to org.apache.flink.table.catalog.hive.util.HiveTableUtils for more details
```
5. get input/output format in following order:
```
    // 1. use input/output from storage format (STORED AS)
    // 2. use input/output from table properties
    // 3. use input/output from default file format
```

### Why are the changes needed?

Fix: #9508

### Does this PR introduce _any_ user-facing change?
no

### How was this patch tested?

The patch was tested with both unit and integration tests:

1. **Unit tests**
   - Added tests for `HivePropertiesConverter` to verify:


5. **Integration test**
   - Added an end-to-end test that creates a Hive table via the Gravitino Flink
     connector and could be read by native flink client.

FANNG1 commented on build.gradle.kts:
```diff
@@ -98,7 +98,14 @@ licenseReport {
+  mavenCentral()
+  // Fallback repository for transient Maven Central 403 errors in CI.
+  // This improves build stability without changing dependency versions or semantics.
+  maven {
```
Could you create a separate PR to track this issue?

FANNG1 commented:
@Pranaykarvi  could you fix the CI by `./gradlew :flink-connector:flink:spotlessApply`?

Pranaykarvi commented:
Hi @FANNG1,

I’ve run ```./gradlew :flink-connector:flink:spotlessApply```, removed the generated ```gradlew.bat```, and rebased the branch.

Could you please take another look when you have a chance? Thanks!

FANNG1 commented:
@jerryshao @Pranaykarvi PTAL
